### PR TITLE
KOGITO-6745: Extend VS Code Channel API to enable support to custom implementations

### PR DESCRIPTION
**Pull request by pefernan**

### CODE REVIEW

**Comment by pefernan** [comment]
jenkins restest this please

File: packages/vscode-extension/src/index.ts [hunk 1]
```diff
@@ -64,7 +67,8 @@ export async function startExtension(args: {
     vsCodeNotificationsApi,
     vsCodeJavaCodeCompletionChannelApi,
     args.viewType,
-    vsCodeI18n
+    vsCodeI18n,
+    args?.channelApiProducer
```
Comment (tiagobento) [inline packages/vscode-extension/src/index.ts]:
```suggestion
    args.channelApiProducer
```

File: packages/vscode-extension/src/KogitoEditorChannelApiProducer.ts [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+import { BackendProxy } from "@kie-tools-core/backend/dist/api";
+import { NotificationsApi } from "@kie-tools-core/notifications/dist/api";
+import { JavaCodeCompletionApi } from "@kie-tools-core/vscode-java-code-completion/dist/api";
+import { I18n } from "@kie-tools-core/i18n/dist/core";
+import { VsCodeI18n } from "./i18n";
+import { Uri } from "vscode";
+
+/**
+ * Produces instances of KogitoEditorChannelApi to be used if we want to provide the extension a channel api with
+ * custom features.
```
Comment (tiagobento) [inline packages/vscode-extension/src/KogitoEditorChannelApiProducer.ts]:
```suggestion
 * Produces instances of KogitoEditorChannelApi to be used if we want to provide the extension a Channel API with
 * custom features.
```

**Review by tiagobento — CHANGES_REQUESTED** [review]
I don't know if this is in scope for this PR, but I feel like `startExtension` should receive a generic parameter now, like T extends KogitoEditorChannelApi, and it should return a way to communicate with the Editor that is there (e.g. the envelopeApi). This generic parameter would be extended to the new Producer interface too.

If we want to provide custom features for specific Editors, this PR is adding a good way of providing methods to the Editors, but we're not able to consume them still. Being able to consume a custom Envelope API would also be a good improvement. WDYT?

**Review by tiagobento — APPROVED** [review]

**Review by paulovmr — APPROVED** [review]

**Comment by tiagobento** [comment]
After speaking with Pere, we decided that it's going to be too much for us to have this type of generic behavior right now, so this PR is enough for now.

**Pull request merged**